### PR TITLE
TDD project

### PR DESCRIPTION
**Pull request by Laylo309**
- Create a class called `Solver`.
- Create a method called `factorial` that takes one argument, an integer `N`, and returns the [factorial](https://en.wikipedia.org/wiki/Factorial) for that number. The factorial is the multiplication of all integers from 1 to `N` and has the special case that the factorial of 0 is 1. This method only accepts 0 and positive integers, so if a negative integer is given it should raise an exception.
- Create a method called `reverse` that takes one argument, a string `word`, and returns `word` reversed (e.g. if `word` is `"hello"` it returns `"olleh"`).
- Create a method called `fizzbuzz` that takes one argument, an integer `N`, and returns a string. The returned string is constructed following these rules:
  - When `N` is divisible by 3, return `"fizz"`.
  - When `N` is divisible by 5, return `"buzz"`.
  - When `N` is divisible by 3 and 5, return `"fizzbuzz"`.
  - Any other case, return `N` as a string (e.g. say `N` is `7` then return `"7"`).
- Remember to start with tests for all of the methods described above. Your commit history should prove that you have used TDD.

### CODE REVIEW

File: rspec/solver_spec.rb [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+    it 'solver.factorial(5) => 120' do
+      expect(solver.factorial(5)).to eq 120
+    end
+    it 'solver.factorial(1) => 1' do
+      expect(solver.factorial(1)).to eq 1
+    end
+    it 'solver.factorial(0) => 1' do
+      expect(solver.factorial(0)).to eq 1
+    end
+  end
```
Comment (helman101) [inline rspec/solver_spec.rb]: - [ ] Remember that you should not only have positive cases, please add negative cases too 😄 
    - For example, what should happen when I pass a negative number through this function? 🤔

Comment (helman101) [reply rspec/solver_spec.rb]: - [ ] Please test special cases, like 0, what should happen if I pass 0 through this function? 🤔

**Review by helman101 — CHANGES_REQUESTED** [review]
# STATUS: CHANGES REQUIRED ♻️ 🔴 

Hi @Laylo309 & @nelsinofrancisco  👋, 

Nice job so far with this project 😄, but there are still some changes to do.

## Required Changes ♻️

- [ ] _Check the comments under the review._

Cheers and Happy coding!👏👏👏

gif

If anything is not 100% clear please leave your questions or comments in the PR thread, we will be glad to help you 😄.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: rspec/solver_spec.rb (hunk 1 above)
Comment (nelsinofrancisco) [reply rspec/solver_spec.rb]: I forgot to raise an Exception when N is a negative number, but I was already testing for that case when the number is 0. 

```    
    it 'solver.factorial(0) => 1' do
      expect(solver.factorial(0)).to eq 1
    end
```

**Review by GzimAsani — APPROVED** [review]
# STATUS: APPROVED :100:         
## Thank you for implementing all the project requirements, you've done a great job :trophy: There is nothing else to say other than... it's time to merge it :champagne:  :tada: Congratulations

Cheers and Happy coding! :clap:  :computer:  :wine_glass: 
Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
### Please be reminded to not open a new pull request after you get a review, only use the same one :thought_balloon: 


 **_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**
 
 _As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

**Pull request merged**